test/amoc_api_execution_handler_SUITE.erl

```diff
@@ -35,122 +43,230 @@ end_per_suite(_Config) ->
                         {ok, mocked}
                 end);
+setup_meck(TC) when TC =:= fail_to_stop_scenario_when_not_running;
+                    TC =:= fail_to_add_users_when_not_running;
+                    TC =:= fail_to_remove_users_when_not_running;
+                    TC =:= fail_to_remove_users_when_not_running;
+                    TC =:= fail_to_update_settings_when_not_running;
+                    TC =:= fail_to_start_when_not_idle->
+    meck:expect(amoc_dist, get_state, fun() -> stopped end);
+setup_meck(TC) when TC =:= fail_to_start_non_existing_scenario;
+                    TC =:= fail_to_start_with_invalid_settings ->
+    meck:expect(amoc_dist, get_state, fun() -> idle end);
+setup_meck(fail_to_update_with_invalid_settings)->
+    meck:expect(amoc_dist, get_state, fun() -> running end);
 setup_meck(_TC) ->
     ok.
 
 %% Test cases
 
 start_scenario(_Config) ->
-    RequestBody = jiffy:encode({[{scenario, sample_test}]}),
-    {HttpCode, _Body} = amoc_api_helper:patch("/execution/start", RequestBody),
-    ?assertEqual(200, HttpCode).
+    JsonBody = #{scenario => sample_test},
+    {HttpCode, Body} = amoc_api_helper:patch("/execution/start", JsonBody),
+    ?assertEqual(200, HttpCode),
+    ?assertEqual(#{}, Body).
 
 start_scenario_with_users_and_settings(_Config) ->
-    RequestBody = jiffy:encode({[{scenario, sample_test},
-                                 {users, 10},
-                                 {settings, settings_for_json()}]}),
-    {HttpCode, _Body} = amoc_api_helper:patch("/execution/start", RequestBody),
-    ?assertEqual(200, HttpCode).
+    JsonBody = #{scenario => sample_test, users => 10, settings => json_settings()},
+    {HttpCode, Body} = amoc_api_helper:patch("/execution/start", JsonBody),
+    ?assertEqual(200, HttpCode),
+    ?assertEqual(#{}, Body).
+
+fail_to_start_when_amoc_dist_fails(_Config) ->
+    JsonBody = #{scenario => sample_test, users => 10, settings => json_settings()},
+    {HttpCode, Body} = amoc_api_helper:patch("/execution/start", JsonBody),
+    ?assertEqual(500, HttpCode),
+    ?assertEqual(#{<<"error">> => <<"mocked">>}, Body).
 
 fail_to_start_non_existing_scenario(_Config) ->
-    RequestBody = jiffy:encode({[{scenario, bad_scenario}]}),
-    {HttpCode, _Body} = amoc_api_helper:patch("/execution/start", RequestBody),
-    ?assertEqual(409, HttpCode).
+    JsonBody = #{scenario => bad_scenario},
+    {HttpCode, Body} = amoc_api_helper:patch("/execution/start", JsonBody),
+    ?assertEqual(500, HttpCode),
+    ?assertEqual(#{<<"error">> => <<"no_such_scenario">>}, Body).
 
 fail_to_start_scenario_without_name(_Config) ->
-    RequestBody = jiffy:encode({[{users, 10}]}),
-    {HttpCode, _Body} = amoc_api_helper:patch("/execution/start", RequestBody),
-    ?assertEqual(400, HttpCode).
+    %% JSON body doesn't correspond to the expected schema, amoc_rest
+    %% framework fails to verify it and returns 400 code & an empty body.
+    JsonBody = #{users=> 10},
```

Comment:
```suggestion
    JsonBody = #{users => 10},
```

Reply:
And in many more `users=> 10` below

Reply:
yeah, seems that elvis doesn't catch missing spaces for maps :(